app/controllers/api/v1/rooms_controller.rb

```diff
@@ -5,11 +5,10 @@ class RoomsController < Api::BaseController
       respond_to :json
 
       def index
-        if params[:conference_id].blank?
-          rooms = Room.all
+        if @conference
+          @conference.venue ? (rooms = @conference.venue.rooms) : (rooms = [])
```

Comment:
Changing it as you suggested would lead to a rubocop offence 
[Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison](http://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/ConditionalAssignment)
How about this?
```
rooms = if @conference
      @conference.venue ? @conference.venue.rooms : Room.none
    else
      Room.all
    end
```